diff --git a/proto/private/BUILD.release b/proto/private/BUILD.release
index 4c2a43f..b37d83c 100644
--- a/proto/private/BUILD.release
+++ b/proto/private/BUILD.release
@@ -3,17 +3,7 @@ load("@rules_proto//proto:defs.bzl", "proto_lang_toolchain")
 # Use precompiled binaries where possible.
 alias(
     name = "protoc",
-    actual = select({
-        ":linux-aarch64": "@com_google_protobuf_protoc_linux_aarch64//:protoc",
-        ":linux-ppc": "@com_google_protobuf_protoc_linux_ppc//:protoc",
-        ":linux-s390x": "@com_google_protobuf_protoc_linux_s390_64//:protoc",
-        ":linux-x86_32": "@com_google_protobuf_protoc_linux_x86_32//:protoc",
-        ":linux-x86_64": "@com_google_protobuf_protoc_linux_x86_64//:protoc",
-        ":macos-x86_64": "@com_google_protobuf_protoc_macos_x86_64//:protoc",
-        ":windows-x86_32": "@com_google_protobuf_protoc_windows_x86_32//:protoc",
-        ":windows-x86_64": "@com_google_protobuf_protoc_windows_x86_64//:protoc",
-        "//conditions:default": "@com_github_protocolbuffers_protobuf//:protoc",
-    }),
+    # `rules_proto` is currently using a prebuilt compiler, in which the linker is hardcoded to `/lib64/ld-linux-x86-64.so.2`.
+    # The patch will force `rules_proto` to rebuild `protoc`, so we can link it properly for NixOS.
+    # Note that this solution is not ideal: optimally we would be getting `protoc` from `nixpkgs` and hooking it into `rules_proto`.
+    # See https://github.com/bazelbuild/rules_proto/issues/115 for a discussion on the topic.
+    actual = "@com_github_protocolbuffers_protobuf//:protoc",
     visibility = ["//visibility:public"],
 )
 
